client/client.go

```diff
@@ -501,7 +501,7 @@ func (c *Client) CollectAllAllocs() error {
 func (c *Client) Node() *structs.Node {
 	c.configLock.RLock()
 	defer c.configLock.RUnlock()
-	return c.config.Node
+	return c.configCopy.Node
 }
```

Comment:
There are other public methods in this file that access fields from  config rather than configcopy. - datacenter/region/nodeid etc. None of them are maps, but are other lurking races there as well?

Reply:
Those fields are immutable once the Client is returned, the mutable fields are the attributes and links, which surfaced via the `Node()` and `Fingerprint()` using the same Node object rather than the readonly copy.